File: python/code_examples/03-functions/main.py
def greet(name: str, times: int = 1) -> str:
    return (f"Hello, {name}! " * times).strip()


def sum_all(*args: int) -> int:
    return sum(args)


def merge(**kwargs: str) -> dict:
    return dict(kwargs)


def main() -> None:
    print(greet("Ada"))
    print(sum_all(1, 2, 3))
    print(merge(a="1", b="2"))


if __name__ == "__main__":
    main()

File: python/code_examples/03-functions/main.js
function greet(name, times = 1) {
  return Array(times).fill(`Hello, ${name}!`).join(" ").trim();
}

function sumAll(...args) {
  return args.reduce((a, b) => a + b, 0);
}

function merge(kwargs) {
  return { ...kwargs };
}

function main() {
  console.log(greet("Ada"));
  console.log(sumAll(1, 2, 3));
  console.log(merge({ a: "1", b: "2" }));
}

main();
